registry/client/repository.go

```diff
@@ -604,6 +621,45 @@ func (bs *blobs) Resume(ctx context.Context, id string) (distribution.BlobWriter
 	panic("not implemented")
 }
 
+func (bs *blobs) Mount(ctx context.Context, sourceRepo string, dgst digest.Digest) (distribution.Descriptor, error) {
```

Comment:
I think `sourceRepo` should be `reference.Named` rather than a string. Perhaps the digest should also be incorporated into this argument (in that case, it would be `reference.Canonical`). Not sure if that makes sense here, since canonical references are usually used for identifying manifests, not blobs. @dmcgowan @stevvooe WDYT?